### PR TITLE
Land post-install on the GitHub section; retire legacy sources page; fix light-mode error page

### PR DESCRIPTION
**Pull request by Oluwatobi-Mustapha**
## Summary
Polishes the post-install GitHub experience and removes a stale pre-split page.

- **Redirect after install** → users now land on the dedicated **`/github/repositories`** section instead of the legacy per-project "Connect environment sources" page. (`CompleteGitHubConnector` redirect path.)
- **Retire the legacy monolithic sources page** — `ProductProjectDetailPage` (~2.2k lines, the one-component-for-all-connectors view from before AWS/GitHub/K8s were split) is removed; its route now `LegacyScopedAppRedirect`s to the GitHub section, matching the existing migration used for `findings`/`ai-risks`. Shared finding/repo helpers that lived in that range were preserved.
- **Light-mode fix** — the GitHub connect **callback error page** heading was hardcoded `#ffffff` (invisible on the light panel); it now uses the theme-aware `--app-text` token. Error copy is also more specific/actionable.

## Light-mode sweep notes
I scanned the stylesheet for the same bug class (a `html[data-theme='light']` selector that sets near-white text). The error-panel heading was the one genuine instance on a surface that flips to light. The remaining matches are intentional (white text on `.idt-home-after-stack` featured/dark cards, buttons, badges), so I deliberately did **not** blind-edit them — that would regress legitimate white-on-dark elements. A deeper per-page light-mode QA can be a follow-up if wanted.

## Not included (deliberate)
- Inline anti-flash theme script in `index.html`: would require a CSP hash change (script-src is `'self'`); minor benefit (only non-default-theme users flash, since the base theme is dark). Deferred to avoid weakening/complicating CSP.

## Verification
- Go: `go build` + `go test ./internal/api/...` green (redirect-path test updated).
- Web: `tsc` clean, `npm run build` green (bundle ~35 KB smaller), `npm run test:ci` 411 passing (legacy-page tests removed; legacy-route test updated to assert the redirect).
- Error-page contrast fix verified via the `--app-text` token semantics + build; recommend a quick eyeball in staging since the callback route is auth-gated.

## AI disclosure
- AI-assisted: no

---
## Summary by cubic
After installing the GitHub App, users now land on `/github/repositories` with the connected environment preselected. We retired the legacy sources page, fixed the light‑mode callback screen, restored GitHub management (Enterprise/PAT, scan policies, posture), and added one‑off scans with polling and safer state handling.

- **Refactors**
  - Retired the legacy sources page; `projects/:projectID` now redirects to `github/repositories?environment=<projectID>`, or to `aws/kubernetes/github/connect` when `?source=` is present (environment preserved).
  - Post‑install redirect now returns `/github/repositories?environment=<projectID>`.
  - Repositories page adds polling and one‑off scans (scoped to the GitHub App when available); drafts reset on environment/workspace changes; stale one‑off completions are ignored.
  - Appearance settings cleanup: removed “Use pointer cursors”, “Contrast”, and “Code font size”; merged cards; compacted selects.

- **Bug Fixes**
  - GitHub connect callback error heading now uses `--app-text` in light mode; error copy is more specific.
  - Restored GitHub management on the connect page: inline Enterprise/PAT form, scan policy panel, and repository/org posture; ignored stale GitHub App install starts and PAT saves across environment/workspace scope; guarded scan policy loads/saves/deletes; reset drafts on environment/workspace changes and on load errors.

<sup>Written for commit c285297e2ca65f9975ea3f133023453c5bc7d6d1. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/identrail/identrail/pull/1659?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: internal/api/github_connect.go [hunk 1]
```diff
@@ -578,7 +578,9 @@ func (s *Service) CompleteGitHubConnector(ctx context.Context, request GitHubCon
 	delete(s.githubConnectStates, normalizedState)
 	s.githubConnectMu.Unlock()
 	status := s.toGitHubConnectionStatus(connection)
-	redirectPath := "/app/" + url.PathEscape(connection.TenantID) + "/" + url.PathEscape(connection.WorkspaceID) + "/projects/" + url.PathEscape(connection.ProjectID)
+	// Land on the dedicated GitHub section after install rather than the legacy
+	// per-project source-connections page. The GitHub section is workspace-scoped.
+	redirectPath := "/app/" + url.PathEscape(connection.TenantID) + "/" + url.PathEscape(connection.WorkspaceID) + "/github/repositories"
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/github_connect.go]: Confirmed and fixed in 6c705aea. The redirect now carries the connected project as `?environment=<connection.ProjectID>` (URL-escaped), so the GitHub section opens on the environment that was just connected instead of falling back to the workspace default. Updated the redirect-path test assertion to match.

File: web/src/App.tsx [hunk 2]
```diff
@@ -4780,7 +4779,7 @@ export function RoutedSite() {
             <Route path="workspaces" element={<ProductWorkspacesPage />} />
             {/* Hidden aliases for pre-domain onboarding and callback URLs. */}
             <Route path="projects" element={<ProductProjectsPage />} />
-            <Route path="projects/:projectID" element={<ProductProjectDetailPage />} />
+            <Route path="projects/:projectID" element={<LegacyScopedAppRedirect target="github/repositories" />} />
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/App.tsx]: Good catch — legitimate regression, fixed in 6c705aea. Deleting the legacy page removed the only UI that called `upsertGitHubPATConnector` while the connect page still linked to it. I ported the GitHub Enterprise / PAT form into `ProductGitHubConnectPage` as a collapsible 'GitHub Enterprise fallback' section, repointed both 'Manage Enterprise / PAT' controls to open it (they no longer target the retired `/projects/:id` route), and removed the now-dead `legacyProjectSetupPath` helper. Connect-page tests updated to assert the inline form instead of the legacy link.

[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/App.tsx]: Fixed in 5428b243. The legacy /projects/:projectID redirect now honors source=aws, source=kubernetes, and source=github, routes each to the matching connector entry point, and carries the selected project as environment=<projectID>. I also added redirect coverage for the default, AWS, Kubernetes, and GitHub cases.

File: web/src/App.tsx [hunk 3]
```diff
@@ -4780,7 +4813,7 @@ export function RoutedSite() {
             <Route path="workspaces" element={<ProductWorkspacesPage />} />
             {/* Hidden aliases for pre-domain onboarding and callback URLs. */}
             <Route path="projects" element={<ProductProjectsPage />} />
-            <Route path="projects/:projectID" element={<ProductProjectDetailPage />} />
+            <Route path="projects/:projectID" element={<LegacyProjectSourceRedirect />} />
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/App.tsx:4816]: Fixed in 9680b3f7. The GitHub connect page now includes scan policy management for the selected environment: it loads existing policies, supports editing trigger mode/cadence/limits, and calls the existing upsert/delete scan-policy APIs. I added product-shell coverage for loading, saving, and deleting the project-scoped policy controls.

[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/App.tsx:4816]: Fixed in 06883dd6. The GitHub repositories page now mounts the repository posture surface for connected GitHub App environments: it fetches getGitHubConnectorRepositoryPosture for the selected repository, shows repository and organization posture checks, and supports switching selected repositories. I added product-shell coverage for the restored posture panel and API call.

File: web/src/productShell.tsx [hunk 4]
```diff
@@ -15198,6 +15206,75 @@ export function ProductGitHubConnectPage() {
...
+        selectedEnvironmentID,
+        {
+          limit: 50,
+          sort_by: 'updated_at',
+          sort_order: 'desc'
+        },
+        buildProductAuthContext(scope)
+      );
+      const items = response.items ?? [];
+      setScanPolicies(items);
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/productShell.tsx:15667]: Fixed in d979be38. The GitHub connect scan-policy loader now uses a request token and ignores stale responses when the selected environment changes before an earlier listProjectScanPolicies request finishes. I added a product-shell regression test that switches from production to staging, resolves the staging policy first, then resolves the stale production response and verifies it is not rendered under the new environment.

File: web/src/productShell.tsx [hunk 5]
```diff
@@ -15388,6 +15396,75 @@ export function ProductGitHubConnectPage() {
...
+      setPolicyForm((current) => {
+        if (items.length === 0) {
+          return current;
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/productShell.tsx]: Fixed in 71064978. Empty scan-policy responses now reset the GitHub connect policy editor back to the default blank draft instead of carrying a previous environment policy into the newly selected environment. I also added a regression test that switches from an environment with a policy to an empty environment and verifies the stale policy name/mode are cleared.

File: web/src/productShell.tsx [hunk 6]
```diff
@@ -15385,9 +15403,81 @@ export function ProductGitHubConnectPage() {
...
+          historyLimit: String(selected.history_limit),
+          maxFindings: String(selected.max_findings)
+        };
+      });
+    } catch (error) {
+      if (scanPolicyRequestRef.current !== requestID) {
+        return;
+      }
+      setScanPolicies([]);
+      setScanPolicyError(formatAPIError(error, 'Unable to load scan policies.'));
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/productShell.tsx:15690]: Fixed in b7e61983. The scan-policy loader now resets the policy draft to the default blank form when the current environment policy request fails, so a stale previous-environment policy cannot be saved into the newly selected environment after an error. I also added a regression test that switches from a policy-bearing environment to one whose policy load fails and verifies the old draft is cleared.

File: web/src/productShell.tsx [hunk 7]
```diff
@@ -15475,6 +15566,117 @@ export function ProductGitHubConnectPage() {
...
+        name: policy.name,
+        enabled: policy.enabled,
+        triggerMode: policy.trigger_mode,
+        cron: policy.cron ?? '',
+        maxConcurrentScans: String(policy.max_concurrent_scans),
+        historyLimit: String(policy.history_limit),
+        maxFindings: String(policy.max_findings)
+      });
+      setScanPolicySuccess('Scan policy saved.');
+      await loadScanPolicies();
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/productShell.tsx:15915]: Fixed in 103a4b3d. The scan-policy save path now captures the environment and save request id, bails out if the selected environment changes before upsertProjectScanPolicy returns, and avoids calling the stale loadScanPolicies closure. I added a regression test that starts a save, switches environments, resolves the old save, and verifies no stale success/form state or old-environment reload appears.

File: web/src/productShell.tsx [hunk 8]
```diff
@@ -15621,55 +15827,391 @@ export function ProductGitHubConnectPage() {
...
+                    >
+                      Edit
+                    </button>
+                    <button
+                      type="button"
+                      className="idt-btn idt-btn-ghost"
+                      onClick={() => {
+                        void handleScanPolicyDelete(policy.policy_id);
+                      }}
+                      disabled={policyDeletingID === policy.policy_id}
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/productShell.tsx]: Fixed in 103a4b3d. The policy loader now clears old scan-policy rows as soon as a new environment load starts, disables row actions while scan policies are loading, and the delete path also ignores stale completions after an environment switch. I added regression coverage for stale delete completion and for clearing old rows during a new environment load.

File: web/src/App.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/App.tsx:4816]: Fixed in f4cd42b2. The GitHub repositories page now polls every 8 seconds while the selected repository data contains an active queued/running scan, using the existing data.reload path so the page refreshes until the scan completes. I added a regression test that captures the polling interval and verifies it reloads repository scans while a row is in flight.

File: web/src/productShell.tsx [hunk 9]
```diff
@@ -15475,6 +15572,151 @@ export function ProductGitHubConnectPage() {
...
+      if (!token) {
+        throw new Error('Enter a GitHub personal access token for the self-hosted fallback.');
+      }
+      await apiClient.upsertGitHubPATConnector(
+        {
+          project_id: selectedEnvironmentID,
+          display_name: normalizeValue(patForm.displayName) || undefined,
+          base_url: normalizeValue(patForm.baseURL) || undefined,
+          token,
+          selected_repositories: parseGitHubRepositories(patForm.repositories)
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/productShell.tsx]: Fixed in d296c8b1. The Enterprise/PAT form now resets its draft, errors, and success state when the selected environment changes, and PAT submit captures the original environment/request id so stale completions are ignored if the user switches environments mid-save. I used the safer reset approach rather than keying token drafts per environment so sensitive PAT drafts are not preserved across scopes. Added regression coverage for both pre-submit environment changes and in-flight stale PAT saves.

File: web/src/App.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/App.tsx:4816]: Fixed in 68aa43bd. The GitHub repositories page now restores an advanced one-off scan form for connected environments, even when no repositories are selected. It accepts an arbitrary repository plus scan mode, history limit, and max findings, and forwards those limits to runRepoScan. I kept selected-repository row scans on the existing connector-scoped path and added regression coverage for the no-selected-repository/PAT-style one-off scan path.

File: web/src/productShell.tsx [hunk 10]
```diff
@@ -15742,6 +16395,43 @@ export function ProductGitHubRepositoriesPage() {
...
+      const repository = normalizeValue(oneOffScanForm.repository);
+      if (!repository) {
+        throw new Error('Repository is required.');
+      }
+      const request: RepoScanRequest = {
+        repository,
+        scan_mode: oneOffScanForm.scanMode,
+        history_limit: parsePositiveInteger(oneOffScanForm.historyLimit, 'History limit'),
+        max_findings: parsePositiveInteger(oneOffScanForm.maxFindings, 'Max findings')
+      };
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/productShell.tsx:16663]: Fixed in ae26648c. The one-off scan form now mirrors the selected-repository scan behavior for GitHub App connections by attaching the selected environment project_id and connector_id before calling runRepoScan. PAT/manual connections still keep the unscoped arbitrary repository path. I added regression coverage for the GitHub App one-off path to verify scan mode, limits, project_id, and connector_id are all forwarded.

**Comment by Oluwatobi-Mustapha** [comment]
Fixed in ccfa49ea. The GitHub App install start path now captures the environment and request id before calling startGitHubConnector, resets pending install state when the environment changes, and ignores stale success/error/finally continuations before opening the install URL, storing the fallback URL, or reloading. I added regression coverage for starting an install in one environment, switching to another before the request resolves, and verifying the old install URL is not opened or rendered.

File: web/src/App.tsx [hunk 11]
```diff
@@ -106,6 +105,40 @@ function LegacyScopedAppRedirect({ target }: { target: string }) {
...
+  const workspaceID = params.workspaceID?.trim();
+  const projectID = params.projectID?.trim();
+
+  if (!tenantID || !workspaceID) {
+    return <Navigate to="/app" replace />;
+  }
+
+  const query = new URLSearchParams(location.search);
+  const requestedSource = query.get('source')?.trim().toLowerCase() ?? '';
+  const target = LEGACY_PROJECT_SOURCE_TARGETS[requestedSource] ?? 'github/repositories';
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/App.tsx:128]: Fixed in 939a5aea. The onboarding setup path now preserves the selected connector provider as source=<provider> when Open setup routes into the legacy project path, so AWS and Kubernetes onboarding continue into their own connect pages instead of falling back to GitHub. Added onboarding coverage for AWS and Kubernetes Open setup navigation.

File: web/src/productShell.tsx [hunk 12]
```diff
@@ -15937,6 +16612,49 @@ export function ProductGitHubRepositoriesPage() {
...
+      if (data.connection.provider === 'github_app') {
+        request.project_id = selectedEnvironmentID;
+        if (data.connection.connector_id) {
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/productShell.tsx:16666]: Fixed in 939a5aea. The GitHub repositories page now resets the one-off scan draft when the selected environment changes, invalidates in-flight one-off submissions, and ignores stale completions before showing success, clearing fields, or reloading. Added regression coverage for both draft reset and stale one-off scan completion after an environment switch.

File: web/src/productShell.tsx [hunk 13]
```diff
@@ -15816,41 +16115,411 @@ export function ProductGitHubConnectPage() {
...
+  }, [postureRepository, selectedRepositoriesKey]);
+
+  useEffect(() => {
+    selectedOneOffScanEnvironmentRef.current = selectedEnvironmentID;
+    oneOffScanRequestRef.current += 1;
+    setOneOffScanForm(createDefaultOneOffRepoScanForm());
+    setOneOffScanSubmitting(false);
+    setScanError('');
+    setScanInfo('');
+  }, [selectedEnvironmentID]);
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/productShell.tsx]: Fixed in 45c20e73. The one-off scan reset/guard now keys on both the selected environment and tenant/workspace scope, so a route change to another workspace with the same environment id clears the draft and invalidates any in-flight one-off submission before it can update the new scope. Added regression coverage for switching from workspace-a to workspace-b while a one-off scan is pending.

**Comment by John-Reese01** [comment]
@codex review

**Review by John-Reese01 — DISMISSED** [review]

File: internal/api/github_connect.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply internal/api/github_connect.go]: test reply

File: web/src/productShell.tsx [hunk 14]
```diff
@@ -14811,13 +14847,15 @@ async function listRepoScansForSelectedRepositories(
...
+  if (scanLimit <= 0) {
     return [];
   }
 
   const pageLimit = Math.max(scanLimit, options.pageLimit ?? scanLimit);
   const maxPages = Math.max(1, options.maxPages ?? GITHUB_MAX_SCAN_PAGE_FETCHES);
-  const allowed = new Set(selectedRepositories.map((repository) => canonicalGitHubRepositoryDisplay(repository).toLowerCase()));
+  const allowed = selectedRepositories.length > 0
+    ? new Set(selectedRepositories.map((repository) => canonicalGitHubRepositoryDisplay(repository).toLowerCase()))
+    : null;
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/productShell.tsx:14858]: Addressed: `listRepoScansForSelectedRepositories` returns `[]` when `selectedRepositories` is empty, so unscoped repo scans are no longer rendered for an environment with no allowlist.

File: web/src/productShell.tsx [hunk 15]
```diff
@@ -15580,9 +15610,111 @@ export function ProductGitHubConnectPage() {
...
+        {
+          limit: 50,
+          sort_by: 'updated_at',
+          sort_order: 'desc'
+        },
+        buildProductAuthContext(scope)
+      );
+      if (scanPolicyRequestRef.current !== requestID) {
+        return;
+      }
```
[the remark this answers is not in the document]
Comment (Oluwatobi-Mustapha) [reply web/src/productShell.tsx:15665]: Addressed: scan-policy load state is now guarded by captured `scopedEnvironmentKeyRef` (`requestPolicyScope`) before applying updates, so stale cross-scope responses are ignored.

**Review by John-Reese01 — APPROVED** [review]
